logstash-core/lib/logstash/webserver.rb

```diff
@@ -110,3 +122,11 @@ def start_webserver(port)
     end
   end
 end
+
+# Reopen the puma class to create a scoped STDERR and STDOUT
+# This operation is thread safe since its done at the class level
+# and force JRUBY to flush his classes cache.
+module Puma
```

Comment:
I am +1 on moving them to patches, I would prefer to keep the `NullLogger` in our namespace thought. Its currently only used by puma but its not puma specific.